### PR TITLE
[modules] enable support for clang modules in range-v3

### PR DESCRIPTION
**Pull request by gnzlbg**
This enables support for clang modules in range-v3:

- fixes headers with missing includes
- adds a simple `module.modulemap` file
- adds travis support
- adds cmake support, to use modules just:

`cmake <src_dir> -DRANGES_MODULES=On -DRANGES_LIBCXX_MODULE=<libcxx_include_path>`

I've benchmark these with clang 5.0, and get about a 20-30% compile time reduction. Depending on how much one decides to "fine-tune" the `module.modulemap` file. I've decided to include the simplest `module.modulemap` file possible, because keeping it on sync for 3-4% compile-time reduction is not worth the trouble.

Have a fun weekend.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -257,8 +257,9 @@ install:
   - if [ -n "$GCC_VERSION" -a "$ASAN" == "On" ]; then CXX_FLAGS="${CXX_FLAGS} -fsanitize=address,undefined -fno-omit-frame-pointer"; fi
   - if [ -n "$CLANG_VERSION" ]; then CXX_FLAGS="${CXX_FLAGS} -D__extern_always_inline=inline"; fi
   - if [ "$LIBCXX" == "On" ]; then CXX_FLAGS="${CXX_FLAGS} -stdlib=libc++ -nostdinc++ -cxx-isystem /usr/include/c++/v1/ -Wno-unused-command-line-argument"; fi
-  - if [ "$HEADERS" == "On" ]; then HEADER_CHECK=1; else HEADER_CHECK=0; fi
+  - if [ "$HEADERS" == "On" ]; then HEADER_CHECK=1; else HEADER_CHECK=0; fi
```
Comment (CaseyCarter) [inline .travis.yml]: Excess whitespace at end-of-line here.

Comment (gnzlbg) [inline .travis.yml]: fixed :)

**Comment by ericniebler** [comment]
If it's possible to have comments in a module map file, then please add a license and copyright there. Thanks.

**Comment by gnzlbg** [comment]
@ericniebler added, C99 comments are supported in module map files.

**Comment by ericniebler** [comment]
Thanks!

**Pull request merged**